intrahospital_api/merge_patient.py

```diff
@@ -248,9 +270,45 @@ def move_non_singletons(subrecord_cls, old_parent, new_parent, old_mrn):
             old_subrecord.save()
 
 
+def move_ipc_status(old_patient, new_patient, old_mrn):
+    """
+    IPC status as currently exists is being brought in from upstream.
+    In the future this will change, but for the moment special case IPC Status to
+    use the ipc status with the most recent condition date for the merged patient.
+    """
+    old_ipc_status = old_patient.ipcstatus_set.get()
+    old_ipc_dates = [
+        getattr(old_ipc_status, field) for field in IPC_DATE_FIELDS if getattr(old_ipc_status, field) is not None
+    ]
+
+    new_ipc_status = new_patient.ipcstatus_set.get()
+    new_ipc_dates = [
+        getattr(new_ipc_status, field) for field in IPC_DATE_FIELDS  if getattr(new_ipc_status, field) is not None
+    ]
+
+    # If there are no dates on either the old or new statuses do nothing.
+    if len(old_ipc_dates) == 0 and len(new_ipc_dates) == 0:
```

Comment:
I'm not convinced this behaviour is definitely what we want,
e.g. what if there are comments & / or boolean flags but no dates on one but not the other? 

Also: Did we look at using `Last_Updated` instead of all these date fields ?